chemprop/utils.py

```diff
@@ -145,6 +146,127 @@ def load_checkpoint(path: str,
     return model
 
 
+def overwrite_state_dict(loaded_param_name: str,
+                        model_param_name: str,
+                        loaded_state_dict: collections.OrderedDict,
+                        model_state_dict: collections.OrderedDict,
+                        logger: logging.Logger = None) -> collections.OrderedDict:
+    """
+    Overwrites a given parameter in the current model with the loaded model.
+    :param loaded_param_name: name of parameter in checkpoint model.
+    :param model_param_name: name of parameter in current model.
+    :param loaded_state_dict: state_dict for checkpoint model.
+    :param model_state_dict: state_dict for current model.
+    :param logger: A logger.
+    :return: The updated state_dict for the current model. 
+    """
+    debug = logger.debug if logger is not None else print
+
+    
+    if model_param_name not in model_state_dict:
+        debug(f'Pretrained parameter "{param_name}" cannot be found in model parameters.')
```

Comment:
Would we want to raise a hard error at this point if it happened?

Reply:
I was thinking that could be nice, it's pretty easy to miss these errors. I think this is a warning in the load_checkpoint function though, and I was trying to be consistent with respect to what triggers errors/warnings.

Reply:
Very good then. I hadn't yet gotten to the load_checkpoint file. Having them all in one place and consistent makes a ton of sense.